docs/book/src/images/cluster-topology-controller.plantuml

```diff
@@ -0,0 +1,32 @@
+@startuml
+
+start
+:new ""Cluster"" object;
+if (Check for the presence of the ""spec.topology.class"" field) then (yes)
+    if (Check if ""Cluster"" has ""Infrastructure"" reference) then (yes)
+    else (no)
+        :Create a new infraCluster object using the ""ClusterClass"" template;
+    endif
+
+    if (Check if ""Cluster"" has ""ControlPlane"" reference) then (yes)
+    else (no)
+        :Create a new ""ControlPlane"" object using the ""ClusterClass"" template;
+    endif
+    :Update ""InfrasCluster"" and ""ControlPlane"" reference in ""Cluster""
+    object using the ""ClusterClass"" template;
```

Comment:
nit
```suggestion
    object using the ""ClusterClass"" template;
```
I think we should either drop or replace this part of the sentence. Essentially we create a InfraCluster and/or ControlPlane before and then we update the refs in the Cluster to point to those objects. The ClusterClass is not involved when setting the references.

(just to avoid misunderstandings)

Reply:
got'cha. makes sense